client/src/components/subscription/updateSubscription.js

```diff
@@ -64,9 +95,85 @@ class UpdateSubscription extends React.Component {
     this.handleSubscriptions();
   }
 
-  render() {
-    const { name, nickname, dueDate, reminderDays, amount } = this.state;
+  handleDays(days) {
+    this.setState({ days });
+  }
+
+  parseDueDate() {
```

Comment:
to your point, a lot of the logic in the update component is the same as the create component. what i think  we can do is abstract the "form" portion into a separate component, then just pass in the data if we're updating and keep the data empty/null if we're creating. i can help walk you through this if you need help